frontend/src/AddQuestionDialog.jsx

```diff
@@ -1,170 +1,179 @@
-				<DialogTitle>Add Question</DialogTitle>
-				<DialogContent>
-					<Box direction="column">
-						<Box mb={2}>
-							<DialogContentText>
-								To add a new question, please enter all details
-								below.
-							</DialogContentText>
-						</Box>
-
-						<form
-							onSubmit={(evt) => {
-								handleSubmit(evt);
-							}}
-						>
-							<Stack spacing={2}>
-								<TextField
-									className="textField same-width-textfield"
-									id="Question Title"
-									label="Question Title"
-									variant="filled"
-									inputRef={inputRefTitle}
-								></TextField>
-								<TextField
-									className="textField same-width-textfield"
-									id="Question Category"
-									label="Question Category"
-									variant="filled"
-									inputRef={inputRefCategory}
-								></TextField>
-								<TextField
-									select
-									name="Question Complexity"
-									label="Question Complexity"
-									variant="filled"
-									defaultValue="Easy"
-									inputRef={inputRefComplexity}
-									className="textField same-width-textfield"
-								>
-									{complexityLevels.map((option) => (
-										<MenuItem value={option} key={option}>
-											{option}
-										</MenuItem>
-									))}
-								</TextField>
-								<TextField
+  const storedQuestions = JSON.parse(localStorage.getItem("questions"));
+
+  const questions = storedQuestions !== null ? storedQuestions : [];
+
+  const inputRefTitle = useRef(null);
+  const inputRefCategory = useRef(null);
+  const inputRefComplexity = useRef(null);
+
+  const complexityLevels = ["Easy", "Medium", "Hard"];
+
+  const [open, setOpen] = useState(false);
+  const [description, setDescription] = useState("");
+
+  const handleClickOpen = () => {
+    setOpen(true);
+  };
+
+  const handleClose = () => {
+    setOpen(false);
+  };
+
+  const [isSnackbarOpen, setSnackbarOpen] = useState(false);
+  const [snackbarMessage, setsnackbarMessage] = useState("");
+
+  const handleDuplicateQuestion = () => {
+    setsnackbarMessage("Duplicate question detected!");
+    setSnackbarOpen(true);
+  };
+
+  const handleEmptyInputField = () => {
+    setsnackbarMessage("Missing fields detected!");
+    setSnackbarOpen(true);
+  };
+
+  const handleSnackbarClose = (event, reason) => {
+    if (reason == "clickaway") {
+      return;
+    }
+    setSnackbarOpen(false);
+  };
+
+  const handleSubmit = (evt) => {
+    evt.preventDefault();
+
+    const title = inputRefTitle.current.value;
+    const complexity = inputRefComplexity.current.value;
+    const category = inputRefCategory.current.value;
+
+    const isDuplicateQuestion =
+      questions !== null &&
+      questions.some((question) => question.title === title);
+
+    const isInputFieldEmpty =
+      !title || !complexity || !category || !description;
+
+    if (isDuplicateQuestion) {
+      handleDuplicateQuestion();
+    } else if (isInputFieldEmpty) {
+      handleEmptyInputField();
```

Comment:
i am able to submit the form if i leave the description field blank. i suspect it's because of the value returned by 'description' which is wrapped around in `<p>` tags, despite of inputted value. could you help include error-handling for this case?

Reply:
I tried to add question with empty description and it pops up error message as expected. May I know how to reproduce the error?

Reply:
I see now, when user types something and deletes it, empty description will be submitted. Will fix this.

Reply:
yep, those were the steps i did to get the error. thanks!